packages/core/test/controls/numericInputTests.tsx

```diff
@@ -57,6 +57,12 @@ describe("<NumericInput>", () => {
             const value = component.state().value;
             expect(value).to.equal("1");
         });
+
+        it("sets allowNumericCharactersOnly to true by default", () => {
+            const component = mount(<NumericInput />);
+            const value = component.props().allowNumericCharactersOnly;
+            expect(value).to.be.true;
```

Comment:
In the case of `<NumbericInput />`, to me, it seems more logical to have a test where you test the functionality of the component when you don't pass anything in - that's actually the output you care about, regardless of what the internals do.

In the examples, I think the problem is actually that behaviour z should be tested in the first place. No one checks the tests to see what the component contracts are (or at least I don't). If I break something and the tests don't catch it - we need more tests.